mentions/src/main/kotlin/io/spine/examples/pingh/mentions/GitHubClientProcess.kt

```diff
@@ -42,58 +45,103 @@ import io.spine.server.event.React
-    }
-
+    /**
+     * Starts the process of updating mentions for the user.
+     *
+     * When a mention update is requested for a user, checks whether the previous update
+     * has ended. If this condition is met, the event of the received request to update mentions
+     * is emitted.
+     */
     @Assign
     @Throws(
         CannotStartDataUpdateTooEarly::class,
         MentionsUpdateIsAlreadyInProgress::class,
         UsersGitHubTokenInvalid::class
     )
     internal fun handle(command: UpdateMentionsFromGitHub): MentionsUpdateFromGitHubRequested {
-
-        if (builder().hasWhenStarted()) {
+        if (state().hasWhenStarted()) {
             throw MentionsUpdateIsAlreadyInProgress.newBuilder()
                 .setId(command.id)
                 .build()
         }
-
         builder().setWhenStarted(currentTime())
         return MentionsUpdateFromGitHubRequested.newBuilder()
-            .setId(command.id)
-            .setToken(command.token)
+            .setId(state().id)
             .vBuild()
     }
 
+    /**
+     * Updates user mentions by emitting [UserMentioned] events
+     * and terminates the mention update process.
+     *
+     * @return List of events, where the [UserMentioned] event for each mention comes first,
+     * followed by a single [MentionsUpdateFromGitHubCompleted] event.
+     */
     @React
-    internal fun on(event: MentionsUpdateFromGitHubRequested): MentionsUpdateFromGitHubCompleted {
+    internal fun on(event: MentionsUpdateFromGitHubRequested): List<EventMessage> {
         archived = true
-
-        builder().clearWhenStarted()
-        return MentionsUpdateFromGitHubCompleted.newBuilder()
-            .setId(event.id)
+        val username = state().id.username
+        val token = state().token
+        val mentions = gitHubClientService.fetchMentions(username, token)
+        val userMentionedEvents = createUserMentionedEvents(mentions)
+        val mentionsUpdateFromGitHubCompleted = MentionsUpdateFromGitHubCompleted.newBuilder()
+            .setId(state().id)
             .vBuild()
+        builder().clearWhenStarted()
+        return listOf(
+            *userMentionedEvents.toTypedArray(),
+            mentionsUpdateFromGitHubCompleted
+        )
+    }
+
+    private fun createUserMentionedEvents(gitHubMentions: Set<Mention>): Set<EventMessage> =
+        gitHubMentions
+            .map { mention ->
+                with(UserMentioned.newBuilder()) {
+                    id = MentionId::class.buildBy(
+                        mention.id,
+                        state().id.username
+                    )
+                    whoMentioned = mention.whoMentioned
+                    title = mention.title
+                    whenMentioned = mention.whenMentioned
+                    url = mention.url
+                    vBuild()
+                }
+            }
+            .toSet()
+
+    /**
+     * Sets the method for fetching mentions from GitHub.
```

Comment:
```
Supplies this instance of the process with a service allowing to access GitHub.

It is expected this method is called right after the creation of the process instance.
Otherwise, the process will not be able to function properly.
```